deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/ConfigHelper.java

```diff
@@ -218,20 +222,20 @@ public static TeletraanServiceContext setupContext(TeletraanServiceConfiguration
             context.setAccountAllowList(configuration.getAccountAllowList());
         }
 
-        /*
-        Lastly, let us create the in-process background job executor, all transient, long
-        running background jobs can be handled by this executor
-        Currently we hard coded the parameters as:
-
-        corePoolSize - the number of threads to keep in the pool, even if they are idle, unless allowCoreThreadTimeOut is set
-        maximumPoolSize - the maximum number of threads to allow in the pool
-        keepAliveTime - when the number of threads is greater than the core, this is the maximum time that excess idle threads will wait for new tasks before terminating.
-        unit - the time unit for the keepAliveTime argument
-        workQueue - the queue to use for holding tasks before they are executed. This queue will hold only the Runnable tasks submitted by the execute method.
-        */
-        // TODO make the thread configurable
+        int poolSize = Runtime.getRuntime().availableProcessors();
+        String jobPoolName = "jobPool";
         ExecutorService jobPool =
-                new ThreadPoolExecutor(1, 10, 30, TimeUnit.SECONDS, new LinkedBlockingQueue<>());
+                environment
+                        .lifecycle()
+                        .executorService(jobPoolName)
+                        .minThreads(poolSize * 4)
+                        .maxThreads(poolSize * 8)
+                        .keepAliveTime(Duration.seconds(30))
+                        .workQueue(new ArrayBlockingQueue<>(poolSize * 5000, false))
+                        .shutdownTime(Duration.seconds(30))
+                        .rejectedExecutionHandler(new AbortPolicy())
+                        .build();
+        new ExecutorServiceMetrics(jobPool, jobPoolName, null).bindTo(Metrics.globalRegistry);
```

Comment:
I see the linked PR uses `InstrumentedExecutorService` instead. Would be nice to be consistent, but we could also update in the future if needed

Reply:
Good catch. This is intentional because `InstrumentedExecutorService` is part of dropwizard and metrics are emitted via dropwizard-metrics, which is being replaced by micrometer.